client/signup/steps/woocommerce-install/hooks/use-woop-handling/index.ts

```diff
@@ -0,0 +1,150 @@
+import { FEATURE_WOOP } from '@automattic/calypso-products';
+import { sprintf, __ } from '@wordpress/i18n';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { addQueryArgs } from 'calypso/lib/url';
+import {
+	fetchAutomatedTransferStatusOnce,
+	requestEligibility,
+} from 'calypso/state/automated-transfer/actions';
+import { eligibilityHolds as eligibilityHoldsConstants } from 'calypso/state/automated-transfer/constants';
+import {
+	isFetchingAutomatedTransferStatus,
+	getEligibility,
+	EligibilityData,
+	EligibilityWarning,
+} from 'calypso/state/automated-transfer/selectors';
+import { requestProductsList } from 'calypso/state/products-list/actions';
+import { getProductBySlug } from 'calypso/state/products-list/selectors';
+import hasActiveSiteFeature from 'calypso/state/selectors/has-active-site-feature';
+import hasAvailableSiteFeature from 'calypso/state/selectors/has-available-site-feature';
+import { getSiteDomain } from 'calypso/state/sites/selectors';
+
+const TRANSFERRING_NOT_BLOCKERS = [
+	eligibilityHoldsConstants.NO_BUSINESS_PLAN, // let's redirect to checkout page
+	eligibilityHoldsConstants.TRANSFER_ALREADY_EXISTS, // let's handle it in transferring endpoint.
+];
+
+type EligibilityHook = {
+	eligibilityHolds?: string[];
+	eligibilityWarnings?: EligibilityWarning[];
+	isFetching: boolean;
+	wpcomDomain: string | null;
+	stagingDomain: string | null;
+	wpcomSubdomainWarning: EligibilityWarning | undefined;
+	transferringBlockers: string[];
+	hasBlockers: boolean;
+	siteUpgrading: {
+		required: boolean;
+		checkoutUrl: string;
+		checkoutText: string;
+		productName: string;
+		description: string;
+	};
+};
+
+export default function useEligibility( siteId: number ): EligibilityHook {
+	const dispatch = useDispatch();
+
+	// Request eligibility data.
+	useEffect( () => {
+		if ( ! siteId ) {
+			return;
+		}
+
+		dispatch( fetchAutomatedTransferStatusOnce( siteId ) );
+		dispatch( requestEligibility( siteId ) );
+		dispatch( requestProductsList() );
```

Comment:
Is there some way we could combine these?

Reply:
I'm not sure. We could combine maybe the AT actions but it needs to request the products list too. I think the simpler way to combine them is as it here :-)